runtime/java/src/common/rawlib/RawProcessHandle.java

```diff
@@ -204,7 +205,12 @@ private void waitOnEnd() {
      * <pre>IOVal<ProcessHandle> ::= cmd::String args::[String] io::IO</pre>
      */
     public static NIOVal spawnProcess(StringCatter cmd, ConsCell args, IOToken io) {
-        List<String> full_cmd = ConsCell.toList(args);
+        List<StringCatter> argsList = ConsCell.toList(args);
```

Comment:
`common.javainterop.ConsCellCollection` provides a nicer way to iterate over the elements of a Silver list without converting to an intermediate `ArrayList`:
```java
for (StringCatter arg : new ConsCellCollection<StringCatter>(args)) {
  full_cmd.add(arg.toString);
}
```

Reply:
`full_cmd.addAll(new ConsCellCollection<StringCatter>(args));` too

Reply:
You also need to map `.toString()` over the args

Reply:
whoops yeah, missed that.

Reply:
I changed it to use Lucas's suggestion now.